SubmersedVR/Tweaks/SnapTurning.cs

```diff
@@ -0,0 +1,561 @@
+            __instance.ReplenishOxygen();
+        
+            return false;
+        }
+    }
+#if BZ
+    //Snap turning override
+    [HarmonyPatch(typeof(Hoverbike), nameof(Hoverbike.PhysicsMove))]
+    static class HoverbikeSnapTurn
+    {
+        public static bool Prefix(Hoverbike __instance)
+        {
+            __instance.isPiloting = __instance.GetPilotingCraft();
+            if (__instance.dockedPad)
+            {
+                return false;
+            }
+            if (!__instance.isPiloting)
+            {
+                return false;
+            }
+            __instance.moveDirection = (AvatarInputHandler.main.IsEnabled() ? GameInput.GetMoveDirection() : Vector3.zero);
+            new Vector3(__instance.moveDirection.x, 0f, __instance.moveDirection.z);
+            Vector2 vector = __instance.overflowInput;
+            if (__instance.energyMixin.IsDepleted())
+            {
+                __instance.moveDirection = Vector3.zero;
+            }
+            __instance.appliedThrottle = (__instance.moveDirection != Vector3.zero);
+            float d = __instance.overWater ? (__instance.horizontalDampening / __instance.waterDampening) : __instance.horizontalDampening;
+            __instance.rb.AddTorque(__instance.transform.right * -vector.x * __instance.sidewaysTorque * __instance.verticalDampening, ForceMode.VelocityChange);
+            if(Settings.IsSnowBikeSnapTurningEnabled == false)
+            {
+                __instance.rb.AddTorque(__instance.transform.up * vector.y * __instance.sidewaysTorque * d, ForceMode.VelocityChange);
+            }
+            if(Settings.IsSnowBikeSnapTurningEnabled == true && SnapTurning.hasSnapTurn)
+            {
+                __instance.rb.transform.rotation = Quaternion.Euler( new Vector3( __instance.rb.transform.rotation.eulerAngles.x, __instance.rb.transform.rotation.eulerAngles.y + SnapTurning.snapTurnAngle,  __instance.rb.transform.rotation.eulerAngles.z));
+                SnapTurning.Reset();
+            }
+            Vector3 velocity = __instance.rb.velocity;
+            Vector3 point = __instance.moveDirection;
+            Mathf.Min(1f, point.magnitude);
+            point.y = 0f;
+            point.Normalize();
+            __instance.horizMoveDir = MainCamera.camera.transform.rotation * point;
+            float d2 = __instance.overWater ? (__instance.forwardAccel / __instance.waterDampening) : __instance.forwardAccel;
+            __instance.rb.AddForce(__instance.horizMoveDir * d2);
+           
+            return false;
+        }
+    }
+#endif
+    //This does 3 things
+    //1 
+    //While piloting vehicles, the player's head camera gets locked into position when using XRSettings.enabled = true
+    //While on the hoverbike we dont want to be locked in because we want to eliminate the yaw and pitch to reduce nausea
+    //2 
+    //Disable the last XRSettings.enabled branch by replacing it with false
+    //3
+    //Enable Player Snap Turning
```

Comment:
Thanks for the comments! Maybe I will try doing a [transpiler](https://harmony.pardeike.net/articles/patching-transpiler.html). They can be quite powerfull, but hard to learn.